docs/snippets/dataStorage/datastorage_interface.cpp

```diff
@@ -0,0 +1,44 @@
+#include <alpaka/alpaka.hpp>
+
+int* ptr;
+
+alpaka::LinearizedIdxGenerator otherLinearizedIdxGenerator{alpaka::Vec{0}};
+alpaka::MdSpan otherMdspan{ptr, alpaka::Vec{0}, alpaka::Vec{0}};
+alpaka::View otherView{alpaka::api::host, ptr, alpaka::Vec{0}, alpaka::Vec{0}};
```

Comment:
These both trigger assertion errors in debug mode because the `DataPitches` constructor:
```
assert(pitchBytes.back() == sizeof(value_type));
```
The pitch would have to be 4 for the integer pointers used here.